src/context/TokensBalancesContext/TokensBalancesContextProvider.tsx

```diff
@@ -0,0 +1,64 @@
+import { useChainId } from "lib/chains";
+import { EMPTY_OBJECT } from "lib/objects";
+import noop from "lodash/noop";
+import {
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+export type TokenBalanceUpdate = {
+  balance?: bigint;
+  diff?: bigint;
+};
+
+export type TokensBalancesUpdates = {
+  [tokenAddress: string]: TokenBalanceUpdate | undefined;
+};
+
+type TokensBalancesContextType = {
+  tokensBalancesUpdates: TokensBalancesUpdates;
+  setTokensBalancesUpdates: Dispatch<SetStateAction<TokensBalancesUpdates>>;
+  resetTokensBalancesUpdates: () => void;
+};
+
+const DEFAULT_TOKENS_BALANCES_CONTEXT: TokensBalancesContextType = {
+  tokensBalancesUpdates: EMPTY_OBJECT,
+  setTokensBalancesUpdates: noop,
+  resetTokensBalancesUpdates: noop,
+};
```

Comment:
to avoid empty structure:

```
createContext<TokensBalancesContextType | null> = createContext(null);
```

and then ensure it's not null in the hook, e.x. https://github.com/gmx-io/gmx-interface/blob/30f2df788e8e6cef010130b91c25059e32b8aa08/src/context/GlobalContext/GlobalContextProvider.tsx#L103